jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/compress/DeflateFrameExtension.java

```diff
@@ -53,10 +57,12 @@ int getTailDropMode()
     @Override
     public void onFrame(Frame frame, Callback callback)
     {
+        if (getWebSocketCoreSession().isAutoFragment())
+            LOG.warn("Frame auto-fragmentation must not be used with DeflateFrameExtension");
```

Comment:
Nooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo
Don't do it every frame!  Do it once when the extension is initialized!

Reply:
The value of `getWebSocketCoreSession().isAutoFragment()` could be changed after the Extension is initialised (could be set in onOpen or after) in which case you would never see the warning.

Reply:
And its not a correct configuration it will just break as soon as a frame is auto-fragmented

Reply:
Hmmmm this still feels wrong.   We should be able to detect bad configuration combinations at initialisation rather than on first frame!

Reply:
Can `.setAutoFragment(true)` trigger the error/warning if `permessage-deflate` is actively negotiated?

Reply:
autoFragment defaults to true so `.setAutoFragment(true)` may not be called but the value of it is still true